spark/src/main/scala/io/substrait/spark/logical/ToLogicalPlan.scala

```diff
@@ -249,12 +249,21 @@ class ToLogicalPlan(spark: SparkSession) extends DefaultRelVisitor[LogicalPlan]
       case a: Aggregate => (a.aggregateExpressions, false)
       case other => (other.output, true)
     }
+    val names = if (project.getHint.isPresent) {
+      project.getHint.get().getOutputNames.asScala
```

Comment:
is the `hint.outputNames` supposed to be just the column names, or also the possible inner struct field names in DFS form (like the other Substrait "names" fields)?

Reply:
It's just the column names, used to distinguish different aliases to the same underlying expression.  Otherwise the Spark optimiser de-duplicates them causing the round-trip equality check to fail (even though they are equivalent).

Reply:
The [docstring](https://github.com/substrait-io/substrait/blob/4a472447392d9beba0401987b4d845079241c30d/proto/substrait/algebra.proto#L44) there says
> // Assigns alternative output field names for any relation.  Equivalent to the names field
> // in RelRoot but applies to the output of the relation this RelCommon is attached to.

There's multiple ways to infer that, but "equivalent to the names field in RelRoot" would indicate it should be the DFS listing, including the inner names, since that's what the "names" field in RelRoot is?

(Using it as the DFS would help with the named_struct issue too ;))

Reply:
Hi @Blizzara, I'm not really sure what you want me to do here.  Do you have a test case in mind that demonstrates the problem?  Thanks :)

Reply:
We should have the "names" hint contain all of the names, including the inner names in the schema, so that it matches the "names" field on the RelRoot. You can see an example of how I'm planning on doing it for RelRoot here: https://github.com/substrait-io/substrait-java/pull/342 (I should merge that but it fails for the the test added in #346  as it needs a fix from https://github.com/substrait-io/substrait-java/pull/315 - it's actually a good catch from that test.)